MainClass.cs

```diff
@@ -0,0 +1,259 @@
+			{
+				ModHelper.Console.WriteLine("Hit Keypad4 or Alpha4", MessageType.Debug);
+				FreeCam.transform.parent = Locator.GetAstroObject(AstroObject.Name.TowerTwin).gameObject.transform;
+			}
+
+			if (Input.GetKeyDown(KeyCode.Keypad5) || Input.GetKeyDown(KeyCode.Alpha5))
+			{
+				ModHelper.Console.WriteLine("Hit Keypad5 or Alpha5", MessageType.Debug);
+				FreeCam.transform.parent = Locator.GetAstroObject(AstroObject.Name.TimberHearth).gameObject.transform;
+			}
+
+			if (Input.GetKeyDown(KeyCode.Keypad6) || Input.GetKeyDown(KeyCode.Alpha6))
+			{
+				ModHelper.Console.WriteLine("Hit Keypad6 or Alpha6", MessageType.Debug);
+				FreeCam.transform.parent = Locator.GetAstroObject(AstroObject.Name.BrittleHollow).gameObject.transform;
+			}
+
+			if (Input.GetKeyDown(KeyCode.Keypad7) || Input.GetKeyDown(KeyCode.Alpha7))
+			{
+				ModHelper.Console.WriteLine("Hit Keypad7 or Alpha7", MessageType.Debug);
+				FreeCam.transform.parent = Locator.GetAstroObject(AstroObject.Name.GiantsDeep).gameObject.transform;
+			}
+
+			if (Input.GetKeyDown(KeyCode.Keypad8) || Input.GetKeyDown(KeyCode.Alpha8))
+			{
+				ModHelper.Console.WriteLine("Hit Keypad8 or Alpha8", MessageType.Debug);
+				FreeCam.transform.parent = Locator.GetAstroObject(AstroObject.Name.DarkBramble).gameObject.transform;
+			}
+
+			if (Input.GetKeyDown(KeyCode.Keypad9) || Input.GetKeyDown(KeyCode.Alpha9))
+			{
+				ModHelper.Console.WriteLine("Hit Keypad9 or Alpha9", MessageType.Debug);
+				FreeCam.transform.position = Locator.GetPlayerTransform().position;
+			}
+
+			if (Input.GetKeyDown(KeyCode.KeypadPlus) || Input.GetKeyDown(KeyCode.Plus))
+			{
+				ModHelper.Console.WriteLine("Hit KeypadPlus or Plus", MessageType.Debug);
+				MoveSpeed = 7f;
+			}
+
+			if (Input.GetKeyDown(KeyCode.KeypadEnter) || Input.GetKeyDown(KeyCode.Return))
+			{
+				ModHelper.Console.WriteLine("Hit KeypadEnter or Return", MessageType.Debug);
+				MoveSpeed = 1000f;
+			}
+
+			if (Input.GetKeyDown(KeyCode.KeypadPeriod) || Input.GetKeyDown(KeyCode.Period))
+			{
+				ModHelper.Console.WriteLine("Hit KeypadPeriod or Period", MessageType.Debug);
+				ToggleMode();
+			}
+		}
+
+		private void ToggleMode()
+		{
+			if (_mode)
+			{
+				_mode = false;
+				if (_storedMode == InputMode.None)
+				{
+					_storedMode = InputMode.Character;
+				}
+
+				OWInput.ChangeInputMode(_storedMode);
+				GlobalMessenger<OWCamera>.FireEvent("SwitchActiveCamera", Locator.GetPlayerCamera());
+				Camera.enabled = false;
+				Locator.GetActiveCamera().mainCamera.enabled = true;
+			}
+			else
+			{
+				_mode = true;
+				_storedMode = OWInput.GetInputMode();
+				OWInput.ChangeInputMode(InputMode.None);
+				GlobalMessenger<OWCamera>.FireEvent("SwitchActiveCamera", _owCamera);
+				Locator.GetActiveCamera().mainCamera.enabled = false;
+				Camera.enabled = true;
+			}
+		}
+
+		public static void MNActivateInput() => InputEnabled = true;
+
+		public static void MNDeactivateInput() => InputEnabled = false;
```

Comment:
ughhh that input handler 🥴 kinda why i abandoned doing these mods - too many separate things to worry about